### PR TITLE
Retry commits in consumeWith when failing due to a RetriableCommitFailedException 

### PR DESCRIPTION
**Pull request by svroonland**
Implements #42 

Adds a `commitOrRetry` method to `Offset` and `OffsetBatch` so it can be used by clients that do not use `consumeWith`

### CODE REVIEW

**Comment by svroonland** [comment]
Suggestion how to fix the doc issue thing? It's either this or a unused import warning as far as I can see.

File: src/main/scala/zio/kafka/client/Offset.scala [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+   * with a [[org.apache.kafka.clients.consumer.RetriableCommitFailedException]]
+   */
+  def commitOrRetry[R, B](policy: Schedule[R, Any, B]): ZIO[R with Clock, Throwable, Unit] =
+    Offset.commitOrRetry(commit, policy)
+}
+
+object Offset {
+  private[client] def commitOrRetry[R, B](
+    commit: Task[Unit],
+    policy: Schedule[R, Any, B]
```
Comment (iravid) [inline src/main/scala/zio/kafka/client/Offset.scala]: If you make this `Schedule[R, Throwable, B]`, the caller will be able to provide a schedule that inspects the Throwable. Is that useful?

**Comment by iravid** [comment]
I'm not that familiar with scaladoc so not sure what the right thing to do there is. You can just drop the `[[ ]]` for now.

File: src/main/scala/zio/kafka/client/Offset.scala (hunk 1 above)
Comment (svroonland) [reply src/main/scala/zio/kafka/client/Offset.scala]: Yeah I suppose so, if you want to retry only on some subtypes of `RetriableCommitException`

File: src/main/scala/zio/kafka/client/Offset.scala [hunk 2]
```diff
@@ -1,13 +1,35 @@
...
   def topicPartition: TopicPartition
   def offset: Long
   def commit: Task[Unit]
   def batch: OffsetBatch
+
+  /**
+   * Attempts to commit and retries according to the given policy when the commit fails
+   * with a RetriableCommitFailedException
+   */
+  def commitOrRetry[R, B](policy: Schedule[R, Throwable, B]): ZIO[R with Clock, Throwable, Unit] =
```
Comment (iravid) [inline src/main/scala/zio/kafka/client/Offset.scala]:
```suggestion
  def commitOrRetry[R](policy: Schedule[R, Throwable, Any]): ZIO[R with Clock, Throwable, Unit] =
```

File: src/main/scala/zio/kafka/client/OffsetBatch.scala [hunk 3]
```diff
@@ -1,13 +1,21 @@
...
   def offsets: Map[TopicPartition, Long]
   def commit: Task[Unit]
   def merge(offset: Offset): OffsetBatch
   def merge(offsets: OffsetBatch): OffsetBatch
+
+  /**
+   * Attempts to commit and retries according to the given policy when the commit fails
+   * with a RetriableCommitFailedException
+   */
+  def commitOrRetry[R, B](policy: Schedule[R, Throwable, B]): ZIO[R with Clock, Throwable, Unit] =
```
Comment (iravid) [inline src/main/scala/zio/kafka/client/OffsetBatch.scala]:
```suggestion
  def commitOrRetry[R](policy: Schedule[R, Throwable, Any]): ZIO[R with Clock, Throwable, Unit] =
```

**Pull request merged**